oidc_auth/util.py

```diff
@@ -1,44 +1,26 @@
-from collections import deque
-import time
-import threading
+import functools
+
+from django.core.cache import caches
+from .settings import api_settings
 
 
 class cache(object):
     """ Cache decorator that memoizes the return value of a method for some time.
     """
+    cache_version = 1
```

Comment:
Just wondering, what does the `cache_version` do?

Reply:
It makes it easier to later change the form of what's being cached without either having to do full cache invalidation on upgrade. I can't think of what kind of changes might be made to the code base in future that would need the cache version changing, but it's there just in case. There's more about it [in the Django docs](https://docs.djangoproject.com/en/2.2/topics/cache/#cache-versioning).

In retrospect, it might be worth making it an optional argument as e.g. `@cache(ttl=…, version=2)` with a default of 1, and a comment/docstring explaining that if you change a cached method to return something incompatible with previous versions, then it needs increasing?